build.gradle.kts

```diff
@@ -188,35 +189,44 @@ tasks.register<Sync>("resolve-default") {
     into(layout.buildDirectory.dir("${project.properties["lib.dir"]}"))
 }
 
+tasks.register<Sync>("resolve-test") {
+    description = "Resolve and retrieve dependencies"
+    dependsOn("clean-test-lib","init-nutch","copy-libs")
+    from(configurations.testCompileClasspath)
+    from(configurations.testRuntimeClasspath)
+    into(layout.buildDirectory.dir("${project.properties["test.lib.dir"]}"))
+}
+
 tasks.register("compile") {
     description = "Compile all Java files"
     dependsOn("compile-core","compile-plugins")
 }
 
 tasks.register<JavaCompile>("compile-core") {
     description = "Compile core Java files only"
-    dependsOn("init-nutch","resolve-default")
+    dependsOn("init-nutch","resolve-default","compileJava")
```

Comment:
Is this necessary? What does `compileJava` do?

Reply:
This dependency is necessary to fix the issue you noted that previously came up when running gradle clean jar. I am not sure if there is an alternate way to fix it, but this explicit declaration seems to work in this case.